### PR TITLE
Fix for distributed training

### PR DESCRIPTION
**Pull request by VoVAllen**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: examples/pytorch/graphsage/experimental/train_dist.py [hunk 1]
```diff
@@ -155,9 +155,27 @@ def evaluate(model, g, inputs, labels, val_nid, test_nid, batch_size, device):
...
+def pad_data(nids):
+    """Repeat nids till same length"""
+    import torch.distributed as dist
+    num_nodes = th.tensor(nids.numel())
+    dist.all_reduce(num_nodes, dist.ReduceOp.MAX)
+    max_num_nodes = int(num_nodes)
+    nids_length = nids.shape[0]
+    if max_num_nodes > nids_length:
+        pad_size = max_num_nodes - nids_length
+        new_nids = th.cat([nids, nids[:pad_size]], axis=0)
```
Comment (Rhett-Ying) [inline examples/pytorch/graphsage/experimental/train_dist.py:167]: is it possible `pad_size > nids_length` then `len(new_nids) < max_num_nodes`?

File: examples/pytorch/graphsage/experimental/train_dist.py [hunk 2]
```diff
@@ -155,9 +155,27 @@ def evaluate(model, g, inputs, labels, val_nid, test_nid, batch_size, device):
     model.train()
     return compute_acc(pred[val_nid], labels[val_nid]), compute_acc(pred[test_nid], labels[test_nid])
 
+def pad_data(nids):
+    """Repeat nids till same length"""
```
Comment (jermainewang) [inline examples/pytorch/graphsage/experimental/train_dist.py:159]: Please elaborate on what will happen if without this function.

**Review by Rhett-Ying — APPROVED** [review]

File: examples/pytorch/graphsage/experimental/train_dist.py [hunk 3]
```diff
@@ -156,19 +156,27 @@ def evaluate(model, g, inputs, labels, val_nid, test_nid, batch_size, device):
...
     num_nodes = th.tensor(nids.numel())
     dist.all_reduce(num_nodes, dist.ReduceOp.MAX)
     max_num_nodes = int(num_nodes)
     nids_length = nids.shape[0]
     if max_num_nodes > nids_length:
-        pad_size = max_num_nodes - nids_length
-        new_nids = th.cat([nids, nids[:pad_size]], axis=0)
+        pad_size = max_num_nodes % nids_length
+        repeat_size = max_num_nodes // nids_length
+        new_nids = th.cat([nids for _ in repeat_size] + [nids[:pad_size]], axis=0)
```
Comment (PhilMurzynowski) [inline examples/pytorch/graphsage/experimental/train_dist.py:175]: Can't iterate over Int `repeat_size`, perhaps you wanted `range(repeat_size)`?

**Comment by VoVAllen** [comment]
@PhilMurzynowski My bad. Thanks for the notice!

**Pull request merged**